vavr/src/main/java/io/vavr/Lazy.java

```diff
@@ -112,9 +112,10 @@ private Lazy(Supplier<? extends T> supplier) {
      * @return A lazy sequence of values.
      * @throws NullPointerException if values is null
      */
+    @SuppressWarnings("Convert2MethodRef") // TODO should be fixed in JDK 9 and Idea
     public static <T> Lazy<Seq<T>> sequence(Iterable<? extends Lazy<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        return Lazy.of(() -> Vector.ofAll(values).map(Lazy::get));
+        return Lazy.of(() -> Vector.ofAll(values).map(lazy -> lazy.get()));
```

Comment:
Does a java 9 build not compile with the method reference?

Reply:
Yeah, it's weird :/
We should probably open a bug for the `JDK` and for `Idea`